hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java

```diff
@@ -0,0 +1,422 @@
+       * keys should be done or not.
+       *
+       * @return true if we should continue iteration of keys, false otherwise.
+       * @throws IOException
+       */
+    abstract boolean processKeyPath(List<String> keyPathList)
+          throws IOException;
+
+      /**
+       * Iterates through all the keys prefixed with the input path's key and
+       * processes the key though processKey().
+       * If for any key, the processKey() returns false, then the iteration is
+       * stopped and returned with false indicating that all the keys could not
+       * be processed successfully.
+       *
+       * @return true if all keys are processed successfully, false otherwise.
+       * @throws IOException
+       */
+    boolean iterate() throws IOException {
+      LOG.trace("Iterating path: {}", path);
+      List<String> keyPathList = new ArrayList<>();
+      //hardcoded make it read from conf
+      if (status.isDirectory()) {
+        LOG.trace("Iterating directory: {}", pathKey);
+        OFSPath ofsPath = new OFSPath(pathKey);
+        String ofsPathprefix =
+            ofsPath.getNonKeyPathNoPrefixDelim() + OZONE_URI_DELIMITER;
+        while (keyIterator.hasNext()) {
+          Table.KeyValue< String, OmKeyInfo > kv = keyIterator.next();
+          String keyPath = ofsPathprefix + kv.getValue().getKeyName();
+          LOG.trace("iterating key path: {}", keyPath);
+          if (!kv.getValue().getKeyName().equals("")
+              && kv.getKey().startsWith("/" + pathKey)) {
+            keyPathList.add(keyPath);
+          }
+          if (keyPathList.size() >= OZONE_FS_ITERATE_BATCH_SIZE) {
+            if (!processKeyPath(keyPathList)) {
+              return false;
+            } else {
+              keyPathList.clear();
+            }
+          }
+        }
+        if (keyPathList.size() > 0) {
+          if (!processKeyPath(keyPathList)) {
+            return false;
+          }
+        }
+        return true;
+      } else {
+        LOG.trace("iterating file: {}", path);
+        keyPathList.add(pathKey);
+        return processKeyPath(keyPathList);
+      }
+    }
+
+    FileStatus getStatus() {
+      return status;
+    }
+  }
+
+  private class RenameIterator extends OzoneListingIterator {
+    private final String srcPath;
+    private final String dstPath;
+
+    RenameIterator(Path srcPath, Path dstPath)
+        throws IOException {
+      super(srcPath);
+      this.srcPath = pathToKey(srcPath);
+      this.dstPath = pathToKey(dstPath);
+      LOG.trace("rename from:{} to:{}", this.srcPath, this.dstPath);
+    }
+
+    @Override
+    boolean processKeyPath(List<String> keyPathList) {
+      for (String keyPath : keyPathList) {
+        String newPath = dstPath.concat(keyPath.substring(srcPath.length()));
+        OFSPath src = new OFSPath(keyPath);
+        OFSPath dst = new OFSPath(newPath);
+
+        String volumeName = src.getVolumeName();
+        String bucketName = src.getBucketName();
```

Comment:
Lets have a helper function to convert src,dst to OzoneManagerProtocolProtos.RenameKeyRequest

Reply:
done